src/widgets/helper/ChannelView.cpp

```diff
@@ -103,6 +104,34 @@ namespace {
                                 });
         };
 
+        // Ignore messages containing this emote
+        if (!getSettings()->isIgnoredMessage(emote.getCopyString()))
+        {
+            menu.addAction("Ignore this emote", [name = emote.getCopyString()] {
+                getSettings()->ignoredMessages.append(
+                    IgnorePhrase(name, false, true, "", true));
+            });
+        }
+        else
+        {
+            menu.addAction(
+                "Unignore this emote", [name = emote.getCopyString()] {
+                    auto &ignored_messages =
+                        *getSettings()->ignoredMessages.readOnly();
+                    const auto it = std::find_if(
+                        ignored_messages.cbegin(), ignored_messages.cend(),
+                        [&](const IgnorePhrase &phrase) {
+                            return name == phrase.getPattern();
+                        });
```

Comment:
Three things:

1. We should avoid reusing variable names to avoid confusion, just change `name` in this scope to something like `emoteName`
2. Use regex to avoid situations, where one emote could block another. (`e.g. Blocking `FeelsDankMan` would also block `FeelsDankManLostHisHat`).
3. Use `QString()` instead of `""`

```suggestion
        QString emoteName(emote.getCopyString());
        QString emoteIgnoreRegex("(\\s|^)%1(\\s|$)");
        if (!getSettings()->isIgnoredMessage(emoteIgnoreRegex.arg(emoteName)))
        {
            menu.addAction("Ignore this emote", [emoteName, emoteIgnoreRegex] {
                getSettings()->ignoredMessages.append(
                    IgnorePhrase(emoteIgnoreRegex.arg(emoteName), true, true,
                                 QString(), true));
            });
        }
        else
        {
            menu.addAction(
                "Unignore this emote", [emoteName, emoteIgnoreRegex] {
                    auto &ignored_messages =
                        *getSettings()->ignoredMessages.readOnly();
                    const auto it = std::find_if(
                        ignored_messages.cbegin(), ignored_messages.cend(),
                        [&](const IgnorePhrase &phrase) {
                            return emoteIgnoreRegex.arg(emoteName) ==
                                   phrase.getPattern();
                        });
```

Reply:
1 & 3 a bit nitpicky but if you've written the code for me I've no issues :smile: